build-config/make/crosstool-ng.make

```diff
@@ -43,24 +45,34 @@ crosstool-ng-download: $(CROSSTOOL_NG_DOWNLOAD_STAMP)
 $(CROSSTOOL_NG_DOWNLOAD_STAMP): $(PROJECT_STAMP)
 	$(Q) rm -f $@ && eval $(PROFILE_STAMP)
 	$(Q) echo "==== Getting upstream $(CROSSTOOL_NG_DESC) ===="
-	$(Q) $(SCRIPTDIR)/fetch-package $(DOWNLOADDIR) $(UPSTREAMDIR) \
-		$(CROSSTOOL_NG_TARBALL) $(CROSSTOOL_NG_URLS)
+	$(Q) if [ $(shell awk 'BEGIN{ print "'$(CROSSTOOL_NG_VERSION)'"<="'$(CROSSTOOL_NG_LATEST_RELEASE_VERSION)'" }') -eq 1 ]; then \
+			$(SCRIPTDIR)/fetch-package $(DOWNLOADDIR) $(UPSTREAMDIR) \
+			$(CROSSTOOL_NG_TARBALL) $(CROSSTOOL_NG_URLS); \
+		 fi
 	$(Q) touch $@
 
 SOURCE += $(CROSSTOOL_NG_SOURCE_STAMP)
 crosstool-ng-source: $(CROSSTOOL_NG_SOURCE_STAMP)
 $(CROSSTOOL_NG_SOURCE_STAMP): $(CROSSTOOL_NG_DOWNLOAD_STAMP)
 	$(Q) rm -f $@ && eval $(PROFILE_STAMP)
 	$(Q) echo "==== Extracting upstream $(CROSSTOOL_NG_DESC) ===="
-	$(Q) $(SCRIPTDIR)/extract-package $(CROSSTOOL_NG_BUILD_DIR) $(DOWNLOADDIR)/$(CROSSTOOL_NG_TARBALL)
+	$(Q) if [ $(shell awk 'BEGIN{ print "'$(CROSSTOOL_NG_VERSION)'"<="'$(CROSSTOOL_NG_LATEST_RELEASE_VERSION)'" }') -eq 1 ]; then \
+			$(SCRIPTDIR)/extract-package $(CROSSTOOL_NG_BUILD_DIR) \
+			$(DOWNLOADDIR)/$(CROSSTOOL_NG_TARBALL); \
+	     else \
+			git clone https://github.com/crosstool-ng/crosstool-ng \
+			$(CROSSTOOL_NG_DIR); \
+	     fi	
```

Comment:
I understand what you are doing here, but just grabbing "master" will likely lead to pain and misery.  It will change out from under you at some point and cause trouble.

We have the same problem with the GRUB project -- they do not make releases.  All one can really do is grab a copy of master from some point in time and base your work on that.  I think a similar approach can work here as well.

For example the master branch of crosstool-ng is at commit id 6e7c6165.  We can fetch a tarball based on that commit id like this:

```
$ wget https://github.com/crosstool-ng/crosstool-ng/archive/6e7c6165.tar.gz
$ ls -l 
-rw-r--r-- 1 curt Development 1930561 Jul 25 07:48 6e7c6165.tar.gz
```

Then we say the CROSSTOOL_NG_VERSION is "6e7c6165" and change the URL to point at github instead of crosstool-ng.org.  Also we can upload the tarball to the ONIE mirror.

The point is -- treating everything as a tarball will simplify things.  Having a special case to "git clone" is no longer be needed.